langlib/lang.query/src/main/ballerina/helpers.bal

```diff
@@ -83,7 +84,15 @@ function getStreamFromPipeline(_StreamPipeline pipeline) returns stream<Type, Co
     return pipeline.getStream();
 }
 
-function toArray(stream<Type, CompletionType> strm, Type[] arr) returns Type[]|error {
+function toArray(stream<Type, CompletionType> strm, Type[] arr, boolean isReadOnly) returns Type[]|error {
+    if isReadOnly {
+        Type[] arr2 = [];
+        return createImmutableArray(check updateArray(strm, arr2));
+    }
+    return updateArray(strm, arr);
+}
+
+function updateArray(stream<Type, CompletionType> strm, Type[] arr) returns Type[]|error {
```

Comment:
We can may be rename this method `createArray`, and likewise for table and xml as well.